### DISCUSSION TITLE
Figure brainstorming discussion

### DISCUSSION THREAD

**Discussion by wavingtowaves** [SORTEE-Github-Hackathon/manuscript#115] [General]
Met today with @LunaSare and @emmajhudgins to discuss one of the figures we'll be working on. We would like to use R to create our figures.

Want to bring in @pedrohbraga and @Aariq since we are working on some items related to figures in R that will eventually necessitate some GitHub actions.

### Things to do
- [x] Rob will merge Eric's PR which creates and R folder in our manuscript repo
- [x] Luna creates `scatterblob` RMD file
- [x] Luna creates CSV file with our figure's underlying data and uploads to GitHub repo
- [x] Create R script for figure
- [ ] Line figure script to GitHub action

### Some questions for @pedrohbraga et al.
- Is there an upper limit to GitHub actions? @emmajhudgins did some research and seems like even if we are building multiple figures with some GitHub actions, it
- Multiple groups are interested in creating figures using R code, can we link these up to GitHub action.
- How do we insert the output of the R script that creates a figure into the "content" of the manuscript.

**Comment by Aariq** [comment]
I think it's not going to be too difficult.  There's a github action on the marketplace for [rendering R Markdown files](https://github.com/marketplace/actions/github-action-for-rmarkdown).  If you set the `fig.path` chunk option, or include `ggsave()` calls, I think it will output the files and commit them. 

I think running an R script is even easier.  The action just needs to have a step like: 

```
steps:
... <stuff to install R and R packages>
- name: Run R Scripts
        run: |
          source("R/figures_tables.R")
        shell: Rscript {0} 
- name: Commit files
        run: |
          git config --local user.email "actions@github.com"
          git config --local user.name "GitHub Actions"
          git add --all
          git commit -am "add figures"
          git push 
```

See example here: https://blog--simonpcouch.netlify.app/blog/r-github-actions-commit/

The somewhat annoying part is making sure the action knows what R packages to install.  Probably the easiest way to do this is with [`renv`](https://rstudio.github.io/renv/articles/ci.html).

**Comment by Aariq** [comment]
The other problem to solve is making sure this runs before manubot compiles the manuscript.  I think the easiest way would be to just add steps to the start of the current manubot action.

**Comment by Aguncan** [comment]
We need sequential colors for this [figure](https://docs.google.com/presentation/d/1TOuQlpWMEt-cfWIii1O6YccC2QjQ67LH7i0x5GDouMw/edit#slide=id.p) if we graphed it in R or other software.
In order to decide the colors this [web site](http://eyetracking.upol.cz/color/) create the sequential color schemes for you. We have 14 categories. Therefore we can first set the number of classes then, decide origin and direction colors and set the color distances between each color. 
Then the web site give RGB, HEX or Lab codes of each classes. You can use these codes to create your color scheme for the figure. 
It may be a confusing, however if you try it, you easily understand the logic and use it for own graphs.

**Reply by Aariq** [reply]
If someone made a .csv of the table part, I'd be happy to take a crack at putting it together as a figure using `gt` and `patchwork`.

**Reply by Aguncan** [reply]
Here is the [gist](https://gist.github.com/Aguncan/76961e8faf99fc145341e270f008e110) file of the figure.
Explanation of headings;
color= sequential hex codes of colors from light blue to dark blue
ways= Ways to use GitHub
difficulty = Technical difficulty (scaled from 1 to 3)
example = Example of Uses (note: N/A used for empty cells, we may later fill them)

**Reply by Aguncan** [reply]
As I mentioned previously, we have an unlimited options for colors. Current color scheme is below, we can later decide for the best.

![Capture](https://user-images.githubusercontent.com/18686476/157419412-1ff9d26a-c3bd-48fe-9160-f33ee034c769.JPG)

**Comment by wavingtowaves** [comment]
## Summary of our discussion re: scatterblob on 3/9/2022

Go to cognitive load table and make sure we agree on both the 'ways to use github' as well as 'cognitive load' categories.

### What's the purpose of the figure?
You can do a lot of collab w/ GitHub without being at top of figure
Here are ways you can use GitHub no matter where you are

### Feedback on figure
- Keep in the misc. github uses (e.g., coding clubs and data standards) since this conveys a wider range of uses.
- space out those middle blobs a bit more in the figure `scatterblob_0`
- Cognitive load table can be included in SI
- Add a couple of labels to y-axis 'low technical difficulty' and 'high technical difficulty'
- @LunaSare will offer up some changes to section titles, run by @robcrystalornelas and @emmajhudgins 

@Aguncan and @Aariq we are currently working on this figure. Based on Ali's comment not sure if you have been working on something like this as well. If so, we are more than happy to collaborate on this. We also picked a colorblind friendly color palette. @LunaSare could you add the code for how you selected the `turbo` palette here?

**Reply by LunaSare** [reply]
Cognitive load spreadsheet [here](https://docs.google.com/spreadsheets/d/1OLBR3M0uV3PMSazWUYHgPeX28wllAgS4PIlJXaNu-2o/edit#gid=0). Feel free to modify and add as needed!

Sure @robcrystalornelas!  The code for selecting color friendly palettes I used is:
```
viridis::turbo(n = nrow(scatterblob_data))
 [1] "#30123BFF" "#424AB3FF" "#467EF4FF" "#31AFF5FF" "#18DAC7FF"
 [6] "#38F491FF" "#83FF52FF" "#BDF534FF" "#E9D539FF" "#FEAA33FF"
[11] "#F8721CFF" "#E03F08FF" "#B61C02FF" "#7A0403FF"
```

Where `n` is the number of colors you want.

**Reply by Aguncan** [reply]
Oh I see that. Figure seems to be evolved from the [initial form](https://docs.google.com/presentation/d/1TOuQlpWMEt-cfWIii1O6YccC2QjQ67LH7i0x5GDouMw/edit#slide=id.p). It would be good see the latest version for feedback. But I cannot find the code or figure files.  I am sorry, maybe it is my fault to find them.

**Reply by LunaSare** [reply]
Not your fault @Aguncan ! The figure is somewhat hidden in the scatterblob branch, [here](https://github.com/SORTEE-Github-Hackathon/manuscript/blob/scatterblob/R/scatterblob.svg).

![Scatterblob](https://github.com/SORTEE-Github-Hackathon/manuscript/raw/scatterblob/R/scatterblob.svg)

**Reply by Aguncan** [reply]
It is OK @LunaSare Thank you for your efforts.

**Comment by LunaSare** [comment]
A couple new versions of the figure!

With [sequential cognitive load](https://github.com/SORTEE-Github-Hackathon/manuscript/blob/scatterblob/R/scatterblob_0.pdf) and with ["real" cognitive load](https://github.com/SORTEE-Github-Hackathon/manuscript/blob/scatterblob/R/scatterblob_1.pdf).

**Reply by Aariq** [reply]
I'm curious why you think "citable and stable code" should be so high up?  Maybe it has something to do with what "stable code" means?

**Reply by wavingtowaves** [reply]
👍🏻  Great! I made a comment in PR #146, but briefly, I really like the 'real' cognitive load one (that ends in _1). And agree that depending on how we define 'stable code' we could potentially move that point lower on the difficulty scale.
